### PR TITLE
feat: cleaner design of scheduled rewards

### PR DESCRIPTION
**Pull request by bajtos**
- refactor: consistent format of scheduled rewards
- fix: less space between the rewards and the unit
- feat: show only 6 decimal places in rewards

Originally, I wanted to use 3 decimal places. However, the screenshot from @patrickwoodhead showed a number `0.00031...`, which would be rendered as `0.0`. Therefore, we decided to use 6 decimal places instead.

I could not find any helper in ethers.js BigNumber class that would allow me to specify how many decimal places to show. I implemented the truncation myself using string manipulation.

### Before

Screenshot 2023-11-09 at 10 12 21

### After

Screenshot 2023-11-09 at 10 12 37

### CODE REVIEW

**Comment by bajtos** [comment]
Tests are failing because of an unrelated change, I am fixing them here: https://github.com/filecoin-station/desktop/pull/1093

**Comment by patrickwoodhead** [comment]
Can we do 6 decimal places? I think that makes it more exciting to first time users.

Also, in your before and after, the number of zeroes has changed. Are you converting from FIL to attoFIL?

**Comment by bajtos** [comment]
> Also, in your before and after, the number of zeroes has changed. Are you converting from FIL to attoFIL or something?

Good catch! I noticed this problem too and fixed it in [450f82d](https://github.com/filecoin-station/desktop/pull/1092/commits/450f82d56880876c61cabb7435da3f9fd1c0c20f)

I updated the PR description to mention 6 digits and show new screenshots.

**Comment by bajtos** [comment]
@juliangruber hindsight please 🙏🏻

File: main/wallet.js [hunk 1]
```diff
@@ -84,7 +84,15 @@ function getBalance () {
...
-  return ethers.utils.formatUnits(scheduledRewards, 18)
+  const fullPrecision = ethers.utils.formatUnits(scheduledRewards, 18)
+  const [whole, fraction] = fullPrecision.split('.')
+  if (fraction === undefined) return fullPrecision
+  const truncated = fraction
+    // keep the first 6 digits, discard the rest
+    .slice(0, 6)
+    // remove trailing zeroes
+    .replace(/0+$/, '0')
+  return [whole, truncated].join('.')
```
Comment (juliangruber) [inline main/wallet.js:95]: would be great to add unit tests for this, this kind of function lends itself perfectly (you want to check many edge cases)

Comment (bajtos) [reply main/wallet.js:95]: Great suggestion! 😍 I should have thought about this myself. 🙈 

Here is the PR; the tests helped me to discover a bug.
https://github.com/filecoin-station/desktop/pull/1100
